### PR TITLE
chore: set up ESLint and hook it up to lint-staged

### PR DESCRIPTION
**Pull request by bhongy**
- set up ESLint to work with prettier and typescript using the project `tsconfig`
- enable ESLint on `src` and `test` folders for js, ts, tsx files in `@formspree/core` and `@formspree/react`
- fix lint errors
  - use explicit type import
  - prefer `const` over `let`
- suppress problematic lint errors that require larger refactoring
- leave lint warnings like `@typescript-eslint/no-explicit-any`, `@typescript-eslint/no-unused-vars`, and `react-hooks/exhaustive-deps` to be fixed in the future (keep this PR from being even larger)

_Note: similar to prettier, since we run lint on lint-staged, we're not running it in CI (redundant)._

### CODE REVIEW

File: turbo.json [hunk 1]
```diff
@@ -13,9 +13,7 @@
       "dependsOn": ["^dev"],
       "cache": false
     },
-    "lint": {
-      "outputs": []
-    },
```
Comment (bhongy) [inline turbo.json:18]: `"outputs": []` doesn't do anything here

File: tsconfig.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
     "preserveWatchOutput": true,
     "skipLibCheck": true,
     "strict": true,
-    "target": "ES2015"
+    "target": "ES2015",
+    "verbatimModuleSyntax": true
```
Comment (bhongy) [inline tsconfig.json:19]: https://typescript-eslint.io/blog/consistent-type-imports-and-exports-why-and-how/#verbatim-module-syntax

File: packages/formspree-react/package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "build": "tsup src/index.ts --format esm,cjs --dts --external react --minify",
     "clean": "rm -rf dist && rm -rf node_modules",
     "dev": "tsup src/index.ts --format esm,cjs --dts --external react --sourcemap",
-    "lint": "eslint src/*.ts*",
```
Comment (bhongy) [inline packages/formspree-react/package.json:30]: We should lint `src` and `test` and to include `.js` files too.

File: packages/formspree-core/package.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
     "build": "tsup src/index.ts --format esm,cjs --dts --minify",
     "clean": "rm -rf dist && rm -rf node_modules",
     "dev": "tsup src/index.ts --format esm,cjs --dts --sourcemap",
-    "lint": "eslint src/*.ts*",
+    "lint": "eslint ./src ./test",
```
Comment (bhongy) [inline packages/formspree-core/package.json:31]: We should lint `src` and `test` and to include `.js` files too.

File: .eslintrc.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+root: true
+env:
+  browser: true
```
Comment (bhongy) [inline .eslintrc.yml:3]: provide browser globals to address `no-undef`

File: .eslintrc.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  es6: true
+  jest/globals: true
```
Comment (bhongy) [inline .eslintrc.yml:4]: provide jest globals to address `no-undef`

File: .eslintrc.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+extends:
+  - eslint:recommended
+  - prettier # eslint-config-prettier
+  - plugin:@typescript-eslint/recommended
+plugins:
+  - jest
+rules:
+  '@typescript-eslint/ban-ts-comment': off # temporary disable while addressing existing violations
+  '@typescript-eslint/consistent-type-exports': error
+  '@typescript-eslint/consistent-type-imports': error
```
Comment (bhongy) [inline .eslintrc.yml:14]: Motivation: https://typescript-eslint.io/blog/consistent-type-imports-and-exports-why-and-how

**Review by colevscode — APPROVED** [review]

**Comment by colevscode** [comment]
Just checking: we're still targeting es2015 so this shouldn't change the distribution, right?

**Comment by bhongy** [comment]
> Just checking: we're still targeting es5 so this shouldn't change the distribution, right?

Correct, we don't change the compilation `target` in this PR.

**Pull request merged**